src/renderer/components/swap/Swap.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useCallback, useMemo, useState } from 'react'
+
+import { Asset, assetToString, baseAmount, bn, getValueOfAsset1InAsset2, PoolData } from '@thorchain/asgardex-util'
+import BigNumber from 'bignumber.js'
+import { useIntl } from 'react-intl'
+
+import { PoolDetails } from '../../services/midgard/types'
+import { toPoolData } from '../../services/midgard/utils'
+import AssetInput from '../uielements/assets/assetInput'
+import AssetSelect from '../uielements/assets/assetSelect'
+import Drag from '../uielements/drag'
+import Slider from '../uielements/slider'
+import { CurrencyInfo } from './CurrencyInfo'
+import * as Styled from './Swap.styles'
+
+type SwapProps = {
+  balance?: number
+  availableAssets: { asset: Asset; priceRune: BigNumber }[]
+  sourceAsset: Asset
+  targetAsset: Asset
+  onConfirmSwap: (source: Asset, target: Asset, amount: BigNumber) => void
+  poolDetails?: PoolDetails
+}
+
+export const Swap = ({
+  balance,
+  availableAssets,
+  onConfirmSwap,
+  sourceAsset: _sourceAsset,
+  targetAsset: _targetAsset,
+  poolDetails = []
+}: SwapProps) => {
+  const intl = useIntl()
+  // convert to hash map here instead of using getPoolDetail
+  const poolData: Record<string, PoolData> = useMemo(() => {
+    return poolDetails.reduce((acc, cur) => {
+      if (!cur.asset) {
+        return acc
+      }
+
+      return { ...acc, [cur.asset]: toPoolData(cur) }
+    }, {} as Record<string, PoolData>)
```

Comment:
[suggestion] Extract such helpers (even if these small) into `utils` module or similar and test it. It looks like we can use it at other places, too.